### PR TITLE
Add new send button and first emoji keyboard implementation

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Add the new send button together with a settings option to disable it and still use the return key.
* Replace the typing indicator with a new emoji button.
* Add `EmojiKeyboardViewController`, which displays a collection of Emoji when the emoji button is tapped. 
* This is a first, rough implementation of the emoji selection view controller, it neither includes the functionality to select the skin tone nor all emojis, yet. There also is no way to quickly jump to a section of emojis so far.
* Update a settings enum to be consistent with the Swift 3 naming guidelines.
* Snapshot tests will come in a separate PR.

### CODE REVIEW

**Comment by typfel** [comment]
Some observations from trying the playground build:

- When I send a message I don't see it in the conversation
- Tapping on a emoji in the emoji keyboard sends it immediately, don't think that's desired behaviour.
- When tapping on the emoji button a second time it dismisses the keyboard. I expected it to switch to the normal keyboard.

**Comment by typfel** [comment]
img_3728

**Comment by daehn** [comment]
I will check the 2 & 3 with design. The first one sounds odd, but might be caused by bumping data-model to the version with the notification changes.

**Comment by typfel** [comment]
1 stopped happening after a while and all the sent messages appeared. Very weird.

**Comment by daehn** [comment]
@typfel I just checked the build as well, looks like the observer is not set up to propagate changes yet

**Comment by mikeger** [comment]
@daehn I believe we would like to have this changes in 2.18, whereas if this one would get merged to develop we are going to have a bad time (current develop is going to emerge into 2.17).

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emojis.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
... "🇨🇿", "🇩🇪", "🇩🇬", "🇩🇯", "🇩🇰", "🇩🇲", "🇩🇴", "🇩🇿", "🇪🇦", "🇪🇨", "🇪🇪", "🇪🇬", "🇪🇭", "🇪🇷", "🇪🇸", "🇪🇹", "🇪🇺", "🇫🇮", "🇫🇯", "🇫🇰", "🇫🇲", "🇫🇴", "🇫🇷", "🇬🇦", "🇬🇧", "🇬🇩", "🇬🇪", "🇬🇫", "🇬🇬", "🇬🇭", "🇬🇮", "🇬🇱", "🇬🇲", "🇬🇳", "🇬🇵", "🇬🇶", "🇬🇷", "🇬🇸", "🇬🇹", "🇬🇺", "🇬🇼", "🇬🇾", "🇭🇰", "🇭🇲", "🇭🇳", "🇭🇷", "🇭🇹", "🇭🇺", "🇮🇨", "🇮🇩", "🇮🇪", "🇮🇱", "🇮🇲", "🇮🇳", "🇮🇴", "🇮🇶", "🇮🇷", "🇮🇸", "🇮🇹", "🇯🇪", "🇯🇲", "🇯🇴", "🇯🇵", "🇰🇪", "🇰🇬", "🇰🇭", "🇰🇮", "🇰🇲", "🇰🇳", "🇰🇵", "🇰🇷", "🇰🇼", "🇰🇾", "🇰🇿", "🇱🇦", "🇱🇧", "🇱🇨", "🇱🇮", "🇱🇰", "🇱🇷", "🇱🇸", "🇱🇹", "🇱🇺", "🇱🇻", "🇱🇾", "🇲🇦", "🇲🇨", "🇲🇩", "🇲🇪", "🇲🇫", "🇲🇬", "🇲🇭", "🇲🇰", "🇲🇱", "🇲🇲", "🇲🇳", "🇲🇴", "🇲🇵", "🇲🇶", "🇲🇷", "🇲🇸", "🇲🇹", "🇲🇺", "🇲🇻", "🇲🇼", "🇲🇽", "🇲🇾", "🇲🇿", "🇳🇦", "🇳🇨", "🇳🇪", "🇳🇫", "🇳🇬", "🇳🇮", "🇳🇱", "🇳🇴", "🇳🇵", "🇳🇷", "🇳🇺", "🇳🇿", "🇴🇲", "🇵🇦", "🇵🇪", "🇵🇫", "🇵🇬", "🇵🇭", "🇵🇰", "🇵🇱", "🇵🇲", "🇵🇳", "🇵🇷", "🇵🇸", "🇵🇹", "🇵🇼", "🇵🇾", "🇶🇦", "🇷🇪", "🇷🇴", "🇷🇸", "🇷🇺", "🇷🇼", "🇸🇦", "🇸🇧", "🇸🇨", "🇸🇩", "🇸🇪", "🇸🇬", "🇸🇭", "🇸🇮", "🇸🇯", "🇸🇰", "🇸🇱", "🇸🇲", "🇸🇳", "🇸🇴", "🇸🇷", "🇸🇸", "🇸🇹", "🇸🇻", "🇸🇽", "🇸🇾", "🇸🇿", "🇹🇦", "🇹🇨", "🇹🇩", "🇹🇫", "🇹🇬", "🇹🇭", "🇹🇯", "🇹🇰", "🇹🇱", "🇹🇲", "🇹🇳", "🇹🇴", "🇹🇷", "🇹🇹", "🇹🇻", "🇹🇼", "🇹🇿", "🇺🇦", "🇺🇬", "🇺🇲", "🇺🇸", "🇺🇾", "🇺🇿", "🇻🇦", "🇻🇨", "🇻🇪", "🇻🇬", "🇻🇮", "🇻🇳", "🇻🇺", "🇼🇫", "🇼🇸", "🇽🇰", "🇾🇪", "🇾🇹", "🇿🇦", "🇿🇲", "🇿🇼"]
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emojis.swift]: Have you checked with android team how they are doing it?

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiKeyboardViewController.swift [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            collectionView.edges == view.edges
+        }
+    }
+
+}
+
+extension EmojiKeyboardViewController: UICollectionViewDataSource {
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/EmojiKeyboardViewController.swift]: Yes, as mentioned this is a first rough draft. Design is working on a spec.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emojis.swift (hunk 1 above)
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/Emojis.swift]: Yes, we might be able to reuse their list which already is segmented in sections https://raw.githubusercontent.com/wireapp/wire-android/master/wire-core/src/main/java/com/waz/zclient/utils/Emojis.java

**Comment by daehn** [comment]
@mikeger @typfel The issues (1 & 2) have been fixed in 34149d5. I would suggest that I will open a separate PR with the improved emoji keyboard with the emoji that android uses  (and sections, recent etc.) once it is ready, otherwise this on eight get to big.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.m [hunk 3]
```diff
@@ -119,7 +125,9 @@ @interface ConversationInputBarViewController ()
 @property (nonatomic) IconButton *sketchButton;
 @property (nonatomic) IconButton *pingButton;
 @property (nonatomic) IconButton *locationButton;
-@property (nonatomic) IconButton *gifButton;
+@property (nonatomic) IconButton *sendButton;
+@property (nonatomic) IconButton *emojiButton;
+@property (nonatomic) IconButton *gifButton; // TODO: GIF button has to be setup correctly
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.m]: This is left in there by purpose and will be resolved when the new Giphy implementation is ready.

**Pull request merged**